mm2src/coins_activation/src/utxo_activation/init_utxo_standard_activation.rs

```diff
@@ -54,22 +56,66 @@ impl InitStandaloneCoinActivationOps for UtxoStandardCoin {
         coin_conf: Json,
         activation_request: &Self::ActivationRequest,
         _protocol_info: Self::StandaloneProtocol,
-        _task_handle: &UtxoStandardRpcTaskHandle,
+        task_handle: &UtxoStandardRpcTaskHandle,
     ) -> MmResult<Self, InitUtxoStandardError> {
         let crypto_ctx = CryptoCtx::from_ctx(&ctx)?;
         let priv_key_policy = priv_key_build_policy(&crypto_ctx, activation_request.priv_key_policy);
 
+        let (sync_status_loop_handle, maybe_sync_watcher) =
+            if coin_conf["enable_spv_proof"].as_bool().unwrap_or(false) && !activation_request.mode.is_native() {
+                let (sync_status_notifier, sync_watcher) = channel(1);
+                (
+                    Some(UtxoSyncStatusLoopHandle::new(sync_status_notifier)),
+                    Some(sync_watcher),
+                )
+            } else {
+                (None, None)
+            };
```

Comment:
I think we need to move this implementation to the `UtxoCoinBuilderCommonOps` trait.
One of the main reasons is that `enable_spv_proof` can be used later for QTUM, BCH and other coins.
Another reason is to avoid parsing coins context fields outside of the coin builder.

In this approach, `maybe_sync_watcher` will be able as `UtxoCoinFields::block_headers_status_watcher`.
This will likely require the use of `mpsc` channel instead of `oneshot`.

Reply:
Done